src/appleseedmaya/exporters/shadingnodeexporter.cpp

```diff
@@ -58,6 +58,12 @@ namespace asr = renderer;
 namespace
 {
 
+const char *g_compToColorParamNames[] = {"compR", "compG", "compB"};
+const char *g_compToColorOutputParamName = "comp";
+
+const char *g_colorToCompParamNames[] = {"compR", "compG", "compB"};
```

Comment:
@dictoon that's a good point, excellent point even.
I suppose for readability and since there are 2 shaders, one for compound->components, other for components->compound, separate arrays for each of the 2 shader cases were made.
I can rework this a bit, since the components are the same for I/O diverging only in the suffixes, and compound name is also the same.